pkg/wasm/cache.go

```diff
@@ -136,7 +136,8 @@ func (c *LocalFileCache) Get(downloadURL, checksum string, timeout time.Duration
 		// Get sha256 checksum and check if it is the same as provided one.
 		sha := sha256.Sum256(b)
 		dChecksum = hex.EncodeToString(sha[:])
-		if checksum != "" && dChecksum != checksum {
+		// checksum sent by istiod can be "nil" if not set by user - magic value used to avoid unmarshaling errors
+		if checksum != "" && checksum != "nil" && dChecksum != checksum {
```

Comment:
OCI also validates the checksum if it is not empty: https://github.com/istio/istio/blob/master/pkg/wasm/imagefetcher.go#L83

so maybe better we handle this before entering this switch statement.

Reply:
thanks, good point. I'm moving it to the conversion code